litespi/clkgen.py

```diff
@@ -61,10 +58,9 @@ def __init__(self, pads, device, cnt_width=8, with_ddr=False):
         self.negedge    = negedge    = Signal()
         self.sample     = sample     = Signal()
         self.update     = update     = Signal()
-        self.clk        = clk        = Signal()
         self.en         = en         = Signal()
         cnt             = Signal(cnt_width)
-
+        clk             = Signal()
```

Comment:
FYI -- Calling this signal name clk is a little bit dangerous. Using a logic generated signal to drive the clk input of a flip flop is not something that should be done. Logic should always drive the clock-enable line if I understand correctly?

@enjoy-digital thoughts?

Reply:
If the divided clock is used as data and not to clock a flip flop, it's fine. But we must be sure to we don't have any combinatorial logic on the inputs or outputs to be sure IOs flip flop will be instantiated.